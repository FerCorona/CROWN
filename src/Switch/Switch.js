import React from 'react';
import { v4 as uuidv4 } from 'uuid';

import Label from '../Label';

const Switch = ({
  id,
  isChecked,
  onChange,
  text,
  stateName,
  extraStyle
}) => (
  <div className={`CROWNSwitch ${extraStyle}`}>
    <input
      type='checkbox'
      id={id || uuidv4()}
      className='checkbox colorSwitch'
      defaultChecked={isChecked}
      onChange={() => onChange({
        stateName,
        value: !isChecked
      })}
    />
    <label htmlFor={id} className='switch colorSwitch' />
    {
      text && (
        <div className='Label'>
          <Label size='small' text={text} color='--black-1' />
        </div>
      )
    }
  </div>
);
Switch.defaultProps = {
  onChange: () => { },
  text: null,
  extraStyle: ''
};

export default Switch;