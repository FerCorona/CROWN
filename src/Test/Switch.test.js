import React from 'react';
import renderer from 'react-test-renderer';
import Switch from '../Switch/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Switch id={1234} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});