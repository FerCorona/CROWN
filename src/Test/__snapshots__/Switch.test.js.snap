// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="CROWNSwitch "
>
  <input
    className="checkbox colorSwitch"
    id={1234}
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="switch colorSwitch"
    htmlFor={1234}
  />
</div>
`;
